force-app/main/default/aura/TaskKanban/TaskKanbanController.js

```diff
@@ -0,0 +1,104 @@
+({
+    doInit : function(component,event,helper)
+    {
+        if(component.get("v.recordId")===undefined)
+        {
+            helper.loadContactWithTasks(component).then(result => {
+                component.set("v.contactWithTasks",result);
+            }).catch(error => {
+            });
+		}
+    },
+	onContactFromRDLoaded : function(component,event,helper)
+	{
+        let targetContact = event.getParam("targetContact");
+        if(targetContact !== undefined
+        &&targetContact !== null)
+        {
+            if(targetContact.Id !== "-1")
+            {
+                component.set("v.selectedContact",targetContact);
+                helper.loadTaskOfContact(component,component.get("v.selectedContact")["Id"]).then(result => {
+                    component.set("v.tasksOfContact",result);
+                    helper.parseContactTaskIntoLists(component,result);
+                    
+                }).catch(error => {
+                    helper.displayToast(component,"error","error!",error);
+                });  
+            }
+        }
+	},
+	onContactListChangeHandler : function(component,event,helper)
+	{
+        let cId = component.find("contactSelected").get("v.value");
+       
+		helper.loadTaskOfContact(component,cId).then(result => {
+            component.set("v.selectedContact",helper.getContactSelectChanged(component,cId));
+            helper.parseContactTaskIntoLists(component,result);
+        }).catch(error => {
+           
+        });         
+	},
+	openNewTaskCardHandler : function(component, event, helper){
+        
+        let cId = component.find("contactSelected");
```

Comment:
this kind of logic should be moved into the component's helper.

Reply:
Yes, Moved all components logic to their respective helper. Will add changes to my next commit.